libs/execution_context/ebpf_program.c

```diff
@@ -21,6 +21,13 @@ static size_t _ebpf_program_state_index = MAXUINT64;
 #define EBPF_MAX_HASH_SIZE 128
 #define EBPF_HASH_ALGORITHM L"SHA256"
 
+// The maximum number of iterations to run when executing a BPF_PROG_RUN before yielding to the scheduler. This is to
+// prevent starving other threads on the the CPU, but also allows the measurement of the time spent in the BPF program
+// to be reasonably accurate. The clock resolution is 100ns and the minimum time for a BPF program to run is around
+// 200ns. So 1024 iterations is a large enough that the measurement is accurate, but small enough that the scheduler is
+// not starved. This should equate to around 200us of execution time.
+#define EBPF_TEST_RUN_BATCH_SIZE 1024
```

Comment:
Have you tried this with a BPF program that is close to the max_instruction_count limit?
I'm afraid this may still be too high for such programs.

Reply:
> Have you tried this with a BPF program that is close to the max_instruction_count limit? I'm afraid this may still be too high for such programs.

Naive question: Is there a reason why this value cannot be supplied on a per-program basis, say by an additional run-time parameter? Or read from environment variable which could then be tweaked on a per-run basis?

Reply:
Modified this to 64 and provided reasoning for the selection of that batch size.